verilog/analysis/verilog_linter_configuration_test.cc

```diff
@@ -775,5 +775,22 @@ TEST(RuleBundleTest, ParseRuleBundleSkipComments) {
   }
 }
 
+// ConfigureFromFlags Tests
+TEST(ConifigureFromFlagsTest, Basic) {
+  LinterConfigurationFlags flags = {
+      verilog::RuleSet::kDefault,
+      verilog::RuleBundle(),
+      "-",
+      false,
+      false,
+      ".",
+      "filename"};
+
+  LinterConfiguration config;
+  auto status = config.ConfigureFromFlags(flags);
+
+  EXPECT_TRUE(status.ok());
+}
+
```

Comment:
Good so far.  Can you test to cover some of the other behavior/code paths in `ConfigureFromFlags`?

Reply:
Thanks for the review! I've added some more tests. I don't think we can add unit tests for all of what is being done there without extra refactoring of the code

Reply:
Can you leave a `//TODO:` note around here with what could/should be covered in the next refactoring?

Reply:
Sure added a comment explaining what I meant - basically that if we also read the configuration files in `main()` we will be able to test the configuration form the content of the configuration files too.

Reply:
Isn't the result of reading a configuration file just a rule-set ? Passing around a LinterConfiguration should contain all that. 
If we want to extend the notion of the _flags_ values being set, we should not call this not LinterConfigurationFlags, but LinterConfigurationOptions, so that it is less tied to the notion of flags (also that is a common way to name such things).

Reply:
(for testing reading from files, there is a [ScopedTestFile](https://github.com/google/verible/blob/master/common/util/file_util.h#L59) when this becomes important)